libvast/include/vast/pipeline.hpp

```diff
@@ -282,28 +314,30 @@ class pipeline final : public operator_base {
   template <class Inspector>
   friend auto inspect(Inspector& f, pipeline& x) -> bool {
     if constexpr (Inspector::is_loading) {
-      auto repr = std::string{};
+      auto definition = std::string{};
       if (not f.object(x)
                 .pretty_name("vast.pipeline")
-                .fields(f.field("repr", repr)))
+                .fields(f.field("definition", definition)))
         return false;
-      auto result = pipeline::parse(repr);
+      auto result = pipeline::parse(definition);
       if (not result) {
-        VAST_WARN("failed to parse pipeline '{}': {}", repr, result.error());
+        VAST_WARN("failed to parse pipeline '{}': {}", definition,
+                  result.error());
         return false;
       }
       x = std::move(*result);
       return true;
     } else {
-      auto repr = x.to_string();
+      auto definition = x.to_string();
       return f.object(x)
         .pretty_name("vast.pipeline")
-        .fields(f.field("repr", repr));
+        .fields(f.field("definition", definition));
     }
   }
 
 private:
-  std::vector<operator_ptr> operators_;
+  std::string definition_ = {};
```

Comment:
Out of curiosity, why did you decide to track this as part of the pipeline object instead of some outer component?